### PR TITLE
docs: document minimum Kubernetes version required to run Akri

### LINKED ISSUE #63: DOCS: specify a minimum kubernetes version to work.
**Is your feature request related to a problem? Please describe.**
No one likes it when their 1.14 cluster won't work. Make it clear what DOES work and beyond.

**Is your feature request related to a way you would like Akri extended? Please describe.**
DOCUMENT the minimum version.

**Describe the solution you'd like**
DOCUMENT the minimum version.

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by kate-goldenring**
**What this PR does / why we need it**:
We need to specify the minimum K8s version needed to run Akri. CRDs were not released to general audiences until version [1.16](https://v1-16.docs.kubernetes.io/docs/setup/release/notes/#what-s-new-major-themes), making this our minimum supported version. This adds that note to the user guide.

**Special notes for your reviewer**:
closes #63 

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] added code adheres to standard Rust formatting (`cargo fmt`)
- [ ] code builds properly (`cargo build`)
- [ ] code is free of common mistakes (`cargo clippy`)
- [ ] all Akri tests succeed (`cargo test`)
- [ ] inline documentation builds (`cargo doc`)
- [ ] version has been updated appropriately (`./version.sh`)

### CODE REVIEW

File: docs/user-guide.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ To best understand the benefits of Akri and jump into using it, we recommend you
 To get started using Akri, you must first decide what you want to discover and whether Akri current supports a protocol that can be used to discover resources of that type. To see the list of currently supported protocols, see our [roadmap](./roadmap.md).
 
 ### Setting up your cluster
```
Comment (jiria) [inline docs/user-guide.md:7]: How does a user get to this page? Should we add a link onto the front page README?

**Review by bfjelds — APPROVED** [review]
we might want to bubble this Kubernetes min ver even higher.  i wonder if there is a badge for the README?

File: docs/user-guide.md (hunk 1 above)
Comment (kate-goldenring) [reply docs/user-guide.md:7]: Seems like a good thing to add under the [Documentation](https://github.com/deislabs/akri#documentation) section on the base README. Can do

Comment (jiria) [reply docs/user-guide.md:7]: Aah, sorry, github search did not reveal that. I think the name is fine. But agree with Brian, we should also have it somewhere higher up.

**Pull request merged**